### PR TITLE
fix: minor corrections around FATAL state

### PR DESCRIPTION
**Pull request by toddbaert**
No substantial changes here, just editorial improvements and corrections.

When implementing https://github.com/open-feature/spec/pull/241, I noticed a few oversights (see comments).

### CODE REVIEW

File: specification/sections/05-events.md [hunk 1]
```diff
@@ -229,9 +229,11 @@ The SDK must update it's internal representation of the provider's state accordi
...
+| `PROVIDER_ERROR`                 | `ERROR`/`FATAL`*                                        |
 | `PROVIDER_CONFIGURATION_CHANGED` | N/A (provider remains in its current state)             |
 | `PROVIDER_CONTEXT_CHANGED`       | N/A (only emitted by SDK during context reconciliation) |
 | `PROVIDER_RECONCILING`           | N/A (only emitted by SDK during context reconciliation) |
 
-see: [provider lifecycle management](01-flag-evaluation.md#17-provider-lifecycle-management)
\ No newline at end of file
+\* If the `error code` associated with the error indicates `PROVIDER_FATAL`, the state is set to `FATAL`
+
+see: [provider lifecycle management](01-flag-evaluation.md#17-provider-lifecycle-management), [provider status](../types.md#provider-status) [error codes](../types.md#error-code)
```
Comment (toddbaert) [inline specification/sections/05-events.md:239]: This is mentioned already in [1.7.5](https://github.com/open-feature/spec/blob/ae286cf5192b4e341ea691caee6a4c95582f32ed/specification.json#L327-L331), but worth making clear in this table.

File: specification/types.md [hunk 2]
```diff
@@ -135,6 +135,7 @@ An enumeration of possible provider states.
 | READY        | The provider has been initialized, and is able to reliably resolve flag values.                    |
 | ERROR        | The provider is initialized but is not able to reliably resolve flag values.                       |
 | STALE        | The provider's cached state is no longer valid and may not be up-to-date with the source of truth. |
+| FATAL        | The provider has entered an irrecoverable error state.                                             |
```
Comment (toddbaert) [inline specification/types.md:138]: This is mentioned already in [1.7.1](https://github.com/open-feature/spec/blob/ae286cf5192b4e341ea691caee6a4c95582f32ed/specification.json#L291-L296), but needs inclusion here.

**Review by thomaspoignant — APPROVED** [review]

File: specification/sections/05-events.md [hunk 3]
```diff
@@ -229,9 +229,11 @@ The SDK must update it's internal representation of the provider's state accordi
...
 | `PROVIDER_STALE`                 | `STALE`                                                 |
-| `PROVIDER_ERROR`                 | `ERROR`                                                 |
+| `PROVIDER_ERROR`                 | `ERROR`/`FATAL`*                                        |
 | `PROVIDER_CONFIGURATION_CHANGED` | N/A (provider remains in its current state)             |
 | `PROVIDER_CONTEXT_CHANGED`       | N/A (only emitted by SDK during context reconciliation) |
 | `PROVIDER_RECONCILING`           | N/A (only emitted by SDK during context reconciliation) |
 
-see: [provider lifecycle management](01-flag-evaluation.md#17-provider-lifecycle-management)
\ No newline at end of file
+\* If the `error code` associated with the error indicates `PROVIDER_FATAL`, the state is set to `FATAL`
```
Comment (toddbaert) [inline specification/sections/05-events.md:237]: This is mentioned already in [1.7.5](https://github.com/open-feature/spec/blob/ae286cf5192b4e341ea691caee6a4c95582f32ed/specification.json#L327-L331), but worth making clear in this table.

**Review by beeme1mr — APPROVED** [review]

**Review by lukas-reining — APPROVED** [review]
Makes sense and LGTM!

**Pull request merged**